Neovim failing to build on Darwin and Darwin Hydra?

**Describe the bug**
```
...
[100%] Building C object src/nvim/CMakeFiles/nvim.dir/xdiff/xhistogram.c.o
[100%] Building C object src/nvim/CMakeFiles/nvim.dir/xdiff/xpatience.c.o
[100%] Building C object src/nvim/CMakeFiles/nvim.dir/xdiff/xprepare.c.o
[100%] Building C object src/nvim/CMakeFiles/nvim.dir/xdiff/xutils.c.o
make[2]: *** No rule to make target '/nix/store/nyf0qhx47vad558rcv8bskfkdvs9wwav-luajit-2.1.0-beta3-luv-1.34.1-1/lib/lua/5.1/libluv.dylib', needed by 'bin/nvim'.  Stop.
make[1]: *** [CMakeFiles/Makefile2:5012: src/nvim/CMakeFiles/nvim.dir/all] Error 2
make: *** [Makefile:152: all] Error 2
builder for '/nix/store/bpxyw4r1h2wklcwc6kiisq2n9cps67k9-neovim-unwrapped-0.4.3.drv' failed with exit code 2
```

**To Reproduce**
Steps to reproduce the behavior:
1. Look at https://hydra.nixos.org/job/nixpkgs/trunk/neovim-unwrapped.x86_64-darwin
or
1. `$ nix-env -iA nixpkgs.neovim` on Darwin

**Expected behavior**
I expect neovim to be buildable on Darwin

**Metadata**
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.
```
λ nix-shell -p nix-info --run "nix-info -m"
these paths will be fetched (1.46 MiB download, 7.88 MiB unpacked):
  /nix/store/035ypsc5cwpb0mxg3vis9q6lxp3wvjsl-bash-interactive-4.4-p23
  /nix/store/cgrsbqls5vxc91sm2nsk6j4lrmpx1cdb-bash-interactive-4.4-p23-man
  /nix/store/d53m2iyciyz9g1ya49awgf66gdkfrhci-readline-7.0p5
  /nix/store/f66c7p1j3hxmglcwqvsjghbafqxv7y5a-nix-info
  /nix/store/g9mldq38hm6yxigz90vswv2d57my794j-bash-interactive-4.4-p23-info
  /nix/store/ifb3hl9gpvcg429rlz5a6hx8plf92yww-bash-interactive-4.4-p23-doc
  /nix/store/rfynznfz5ckpvyrdxwhc4sm8q9sikqq8-bash-interactive-4.4-p23-dev
  /nix/store/vsykql6171j54wn2m2bqpjnygxp30dwi-DarwinTools-1
copying path '/nix/store/ifb3hl9gpvcg429rlz5a6hx8plf92yww-bash-interactive-4.4-p23-doc' from 'https://cache.nixos.org'...
copying path '/nix/store/vsykql6171j54wn2m2bqpjnygxp30dwi-DarwinTools-1' from 'https://cache.nixos.org'...
copying path '/nix/store/g9mldq38hm6yxigz90vswv2d57my794j-bash-interactive-4.4-p23-info' from 'https://cache.nixos.org'...
copying path '/nix/store/cgrsbqls5vxc91sm2nsk6j4lrmpx1cdb-bash-interactive-4.4-p23-man' from 'https://cache.nixos.org'...
copying path '/nix/store/f66c7p1j3hxmglcwqvsjghbafqxv7y5a-nix-info' from 'https://cache.nixos.org'...
copying path '/nix/store/d53m2iyciyz9g1ya49awgf66gdkfrhci-readline-7.0p5' from 'https://cache.nixos.org'...
copying path '/nix/store/035ypsc5cwpb0mxg3vis9q6lxp3wvjsl-bash-interactive-4.4-p23' from 'https://cache.nixos.org'...
copying path '/nix/store/rfynznfz5ckpvyrdxwhc4sm8q9sikqq8-bash-interactive-4.4-p23-dev' from 'https://cache.nixos.org'...
 - system: `"x86_64-darwin"`
 - host os: `Darwin 19.3.0, macOS 10.15.3`
 - multi-user?: `no`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.3.1`
 - channels(rpearce): `"home-manager, nixpkgs-20.09pre217039.13e7a3e1127"`
 - nixpkgs: `/Users/rpearce/.nix-defexpr/channels/nixpkgs`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute: neovim
# a list of nixos modules affected by the problem
module:
```

--- a/pkgs/development/lua-modules/generated-packages.nix
+++ b/pkgs/development/lua-modules/generated-packages.nix
@@ -1326,11 +1326,11 @@ luuid = buildLuarocksPackage {
 };
 luv = buildLuarocksPackage {
   pname = "luv";
-  version = "1.34.1-1";
+  version = "1.30.0-0";
 
   src = fetchurl {
-    url    = https://luarocks.org/luv-1.34.1-1.src.rock;
-    sha256 = "044cyp25xn35nj5qp1hx04lfkzrpa6adhqjshq2g7wvbga77p1q0";
+    url    = https://luarocks.org/luv-1.30.0-0.src.rock;
+    sha256 = "1z5sdq9ld4sm5pws9qxpk9cadv9i7ycwad1zwsa57pj67gly11vi";
   };
   disabled = (luaOlder "5.1");
   propagatedBuildInputs = [ lua ];

